### PR TITLE
doc: posix: update for `POSIX_FILE_SYSTEM_R`

### PR DESCRIPTION
**Pull request by ycsin**
Update the notes for `POSIX_FILE_SYSTEM_R`, looks like I missed the v4 release

### CODE REVIEW

**Comment by kartben** [comment]
Ya unfortunately it's too late. You may maybe submit a PR against the branch but I never remember if it's something we do or if submissions *have* to be handled as backports

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -1381,6 +1382,7 @@ Libraries / Subsystems
   * :ref:`_POSIX_THREAD_SAFE_FUNCTIONS <posix_option_thread_safe_functions>` improvements:
 
     * Support for :c:func:`asctime_r`, :c:func:`ctime_r`, and :c:func:`localtime_r`.
+    * Added stub for :c:func:`getpwnam_r`, :c:func:`getpwuid_r`, :c:func:`getgrgid_r` and :c:func:`getgrnam_r`.
```
Comment (cfriedt) [inline doc/releases/release-notes-4.0.rst:1385]: I think this is maybe Ok for a backport PR to v4.0. 

However, looking into things a bit deeper, `getpwnam_r()`, `getpwuid_r()`, `getgrgid_r()`, and `getgrnam_r()` belong to a separate Option Group - 
`POSIX_SYSTEM_DATABASE_R`. And the non-reentrant versions of these belong to the `POSIX_SYSTEM_DATABASE` Option Group, and `_POSIX_THREAD_SAFE_FUNCTIONS` is more of an Option / Feature Test Macro.

So really what we need here is a new `Kconfig.system_database` file with two new Kconfig options (same as the option group names).

`_POSIX_THREAD_SAFE_FUNCTIONS` should probably be a hidden Kconfig option that depends on other Kconfig values being enabled. Namely, `CONFIG_POSIX_SYSTEM_DATABASE_R`, `CONFIG_POSIX_C_LANG_SUPPORT_R`, `CONFIG_POSIX_FILE_LOCKING`, `CONFIG_POSIX_FILE_SYSTEM_R`.

And then of course modify [the docs](https://docs.zephyrproject.org/latest/services/portability/posix/option_groups/index.html#posix-thread-safe-functions)

https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html

Comment (cfriedt) [reply doc/releases/release-notes-4.0.rst:1385]: All of the above should be in a separate PR.

**Comment by dkalowsk** [comment]
Very much missed the release.  Happy to add to the 4.0.1 backport list though!

**Comment by mmahadevan108** [comment]
@ycsin . Do you think you can move the changes from the 4.0 release notes to 4.1 release notes?

**Comment by ycsin** [comment]
> @ycsin . Do you think you can move the changes from the 4.0 release notes to 4.1 release notes?

Should we? The changes were introduced in 4.0, not 4.1.

**Comment by kartben** [comment]
> > @ycsin . Do you think you can move the changes from the 4.0 release notes to 4.1 release notes?
> 
> Should we? The changes were introduced in 4.0, not 4.1.

Yes but it's already been released, so by definition there's no way the release notes can be updated after the fact. They're already there and won't change: https://docs.zephyrproject.org/4.0.0/releases/release-notes-4.0.html
Updating the document in main would *not* be a good solution as we don't want to have discrepancies with the published release notes from the previous link (and released artifacts on GitHub)

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1]: See previous comment

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (ycsin) [reply doc/releases/release-notes-4.0.rst:1]: #76401 hmm...

Comment (kartben) [reply doc/releases/release-notes-4.0.rst:1]: Yup, we discussed setting up some kind of CI check to catch these.

**Comment by ycsin** [comment]
I think I confused GitHub to the point that it won't reflect the changes of the PR branch anymore, recreated a PR without changes to the 4.0 release notes

**Pull request closed without merging**